sgv2-restapi/src/main/java/io/stargate/sgv2/restsvc/resources/CreateGrpcClientFilter.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.restsvc.resources;
+
+import io.stargate.sgv2.common.grpc.StargateBridgeClient;
+import io.stargate.sgv2.common.grpc.StargateBridgeClientFactory;
+import io.stargate.sgv2.restsvc.impl.GrpcClientJerseyFactory;
+import io.stargate.sgv2.restsvc.models.RestServiceError;
+import java.util.Optional;
+import java.util.UUID;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * Validates that the auth token is present and pre-builds the gRPC client before each request.
+ *
+ * <p>Note that the resource class or method must be annotated with {@link CreateGrpcStub} in order
+ * for this filter to run.
+ *
+ * @see GrpcClientJerseyFactory
+ */
+@Provider
+@CreateGrpcStub
+public class CreateGrpcClientFilter implements ContainerRequestFilter {
```

Comment:
Maybe rename to match new names `CreateStargateBridgeClientFilter`? Not a big deal though.